client/gutenberg/extensions/publicize/async-publicize-lib.jsx

```diff
@@ -54,7 +37,9 @@ export function requestPublicizeConnections( postId ) {
  * @return {object} List of possible services that can be connected to
  */
 export function getAllConnections() {
-	return JSON.parse( gutenberg_publicize_setup.allServices );
+	return apiFetch( {
+		path: '/publicize/services',
+	} );
```

Comment:
Aren't we changing the return type here? `JSON.parse()` is sync and returns a string. `apiFetch` is async and returns a promise, doesn't it?

We should probably start using `@wordpress/data` for Publicize -- mostly so we can rely on its resolvers for this kind of data fetch.

Reply:
Ah, I see you've changed the [caller](https://github.com/Automattic/wp-calypso/pull/28050/files#diff-7068bb349869f1a56778e0f6f9dbcaceR26) accordingly.

Okay, let's leave it as is for now (since it makes the entire file quite consistent), but let's at least change the JSDoc `@return`.

As a follow-up, I'd still suggest turning this into a proper `@wordpress/data` based store (including resolvers, to help us get rid of manual promise handling in React lifecycle methods).

Reply:
Oh, turns out there already is a store in `gutenberg-store.jsx` -- it just doesn't have any resolvers yet. So we basically just need to move stuff from `async-publicize-lib.jsx` over there (and rename files).

Reply:
Opened #28100 for now, so we keep track of it.